### PR TITLE
Use removeIf rather than Iterator-based removal

### PR DESCRIPTION
**Pull request by SeasonPanPan**
Hello, I saw several complex and outdated **iterator loops** used in the code while learning the spring boot source code. 

I think this can be optimized for a more concise code. 

Please review it. Thank you for your time.

### CODE REVIEW

File: spring-boot-project/spring-boot-tools/spring-boot-configuration-processor/src/json-shade/java/org/springframework/boot/configurationprocessor/json/JSONArray.java
Comment (wilkinsona) [inline spring-boot-project/spring-boot-tools/spring-boot-configuration-processor/src/json-shade/java/org/springframework/boot/configurationprocessor/json/JSONArray.java:1]: Can you please revert this part of the change. This is a shaded copy of third-party source and we want to leave it as originally written.

**Review by wilkinsona — CHANGES_REQUESTED** [review]
Thanks very much for the PR. I've left a comment for your consideration when you have some time.

**Comment by SeasonPanPan** [comment]
> Thanks very much for the PR. I've left a comment for your consideration when you have some time.

@wilkinsona Done, please check.

**Comment by snicoll** [comment]
@SeasonPanPan thank you for making your first contribution to Spring Boot.

**Pull request closed without merging**